### PR TITLE
TW-1675: presence status updated too many times

### PR DESCRIPTION
**Pull request by Te-Z**
Linked issue: https://github.com/linagora/twake-on-matrix/issues/1675
Linked pull request: https://github.com/linagora/twake-on-matrix/pull/1684

### CODE REVIEW

File: lib/src/client.dart [hunk 1]
```diff
@@ -1801,13 +1805,27 @@ class Client extends MatrixApi {
...
     for (final newPresence in sync.presence ?? []) {
       final cachedPresence = CachedPresence.fromMatrixEvent(newPresence);
       presences[newPresence.senderId] = cachedPresence;
       // ignore: deprecated_member_use_from_same_package
       onPresence.add(newPresence);
       onPresenceChanged.add(cachedPresence);
+
+      if (lowestLastActivePresence == null ||
+          cachedPresence.lastActiveTimestamp!
+              .isBefore(lowestLastActivePresence.lastActiveTimestamp!)) {
```
Comment (sherlockvn) [inline lib/src/client.dart:1820]: `cachedPresence.lastActiveTimestamp` can be null and `lowestLastActivePresence.lastActiveTimestamp` can be null

**Review by hoangdat — COMMENTED** [review]
I still not get the idea for this change, can you explain a little bit @Te-Z

File: lib/src/client.dart [hunk 2]
```diff
@@ -1801,13 +1805,27 @@ class Client extends MatrixApi {
         await _handleRooms(leave, direction: direction);
       }
     }
+
+    CachedPresence? lowestLastActivePresence;
```
[the remark this answers is not in the document]
Comment (Te-Z) [reply lib/src/client.dart:1809]: to update it while the `for each` loop is running (check lines 1811 to 1823). Then I use it to update `onlatestPresenceChanged` line 1826 if it's not null

**Comment by Te-Z** [comment]
> I still not get the idea for this change, can you explain a little bit @Te-Z

Sure @hoangdat 

This was the previous code:
https://github.com/linagora/matrix-dart-sdk/blob/e0b65295255b852532878a948662920d6963cc6f/lib/src/client.dart#L1793-L1799

The UI previously listened to `onPresenceChanged` which is a `CachedStreamController` . This instance had a new value added in the `for each` loop for every items in `sync.presence`list. This means if there's  different values in `sync.presence` , the UI will be updated 6 times.

To handle that I changed for this code:

https://github.com/linagora/matrix-dart-sdk/blob/b0c34117c60360d3f1241af9653ddaaa88aaead4/lib/src/client.dart#L1809-L1829

Here `lowestLastActivePresence` is updated for each items in `sync.presence` list if it is `null` or if the current item's timestamp is previous the one in `lowestLastActivePresence`. Then when the loop is over and we are sure to have the right value, we can update onLatestPresenceChange with the right value and this way update the UI.

**Pull request merged**